### PR TITLE
Update the integrator documentation

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: doc/integrator/cluster.rst [hunk 1]
```diff
@@ -3,36 +3,21 @@
...
-
-A possibility is to have the same database for all servers (integration, production).
-A single schema for static data is shared, whereas a dedicated main schema will
-be used for each server.
+A cluster is a configuration where several servers are used to serve the application in production.
 
 Deploying data from integration to production may be done by
-`cloning the main integration schema <https://wiki.postgresql.org/wiki/Clone_schema>`_.
+`cloning the main integration schema <https://wiki.postgresql.org/wiki/Clone_schema>`_,
+a maintain version of the script is available in ``scripts/CONST_clone_schema.sql``.
```
Comment (jwkaltz) [inline doc/integrator/cluster.rst]: what do we mean by "a maintain version"?

File: doc/integrator/docker.rst [hunk 2]
```diff
@@ -9,225 +9,31 @@ Architecture schema
 .. image:: docker.png
 .. source file is docker.dia.
 
-Configure your project
-----------------------
+Images
+------
 
-Edit the ``<package>.mk`` file and add those lines:
+When we bild the porject it will generate the following images:
```
Comment (jwkaltz) [inline doc/integrator/docker.rst]: "When we build the project..."

File: doc/integrator/docker.rst [hunk 3]
```diff
@@ -9,225 +9,31 @@ Architecture schema
...
-A ``docker-compose.yml.mako`` file is created as a starting point.
-
-If you want to host the database on your local machine, you must add a
-``dbhost`` and ``dbhost_slave`` entry pointing to ``172.17.0.1`` (your host address for Docker
-container) in your ``vars.yaml`` file. Then you need to make sure
-Postgres is configured to listen on that interface and accepts authentication.
-
-If you want to use an external server for the database, just put its address
-in the ``dbhost`` and ``dbhost_slave`` entry.
+``docker-compose.yaml``: The main file that describe the composition.
```
Comment (jwkaltz) [inline doc/integrator/docker.rst]: that describes

File: doc/integrator/docker.rst [hunk 4]
```diff
@@ -9,225 +9,31 @@ Architecture schema
...
+``docker-compose-lib.yaml``: Probide the base description of Geomapfish Docker sevices.
```
Comment (jwkaltz) [inline doc/integrator/docker.rst]: Provide the base description of Geomapfish Docker services.

File: doc/integrator/docker.rst [hunk 5]
```diff
@@ -9,225 +9,31 @@ Architecture schema
...
+``.env``: Build from ``.env.mako`` the the environment variable used by the composition.
```
Comment (jwkaltz) [inline doc/integrator/docker.rst]: "the the" -> "the"

File: doc/integrator/extend_data_model.rst [hunk 6]
```diff
@@ -1,55 +1,245 @@
...
         id = Column(types.Integer, primary_key=True)
-        name = Column(types.Unicode, nullable=False, label=_(u'Name'))
-        description = Column(types.Unicode, label=_(u'Description'))
+        name = Column(types.Unicode, nullable=False, info={
+            'colanderalchemy': {
+                'title': _('Name')
+            }
+        })
+
+And for the admin interface add create the following empty files:
```
Comment (jwkaltz) [inline doc/integrator/extend_data_model.rst]: Suggestion: "Now you need to extend the administration user interface. For this, first ensure that the following files exist (if needed, create them as empty files):"

File: doc/integrator/extend_data_model.rst [hunk 7]
```diff
@@ -1,55 +1,245 @@
...
+
+* ``geoportal/<package>_geoportal/admin/__init__.py``:
+* ``geoportal/<package>_geoportal/admin/view/__init__.py``:
+
+the file ``geoportal/<package>_geoportal/admin/view/userdetail.py``:
```
Comment (jwkaltz) [inline doc/integrator/extend_data_model.rst]: Now, create a file ``geoportal/<package>_geoportal/admin/view/userdetail.py`` as follows:

File: doc/integrator/extend_data_model.rst [hunk 8]
```diff
@@ -1,55 +1,245 @@
...
+
+        @view_config(
+            route_name='c2cgeoform_item_duplicate',
+            request_method='GET',
+            renderer='./templates/edit.jinja2'
+        )
+        def duplicate(self):
+            return super().duplicate()
+
+and finally the file ``geoportal/<package>_geoportal/admin/view/title.py``:
```
Comment (jwkaltz) [inline doc/integrator/extend_data_model.rst]: "and finally" -> "and now"
(we already have a and finally later...)

File: doc/integrator/filter.rst [hunk 9]
```diff
@@ -9,44 +9,7 @@ Filterable layers (public)
...
-
-  - Edit the role that shall be able to use this filter, and activate the corresponding
-    functionality checkbox.
-
-* Note that if there are functionalities associated to a role, then users with this role will
-  not have access to any of the "anonymous" functionalities. Instead, you will need to activate
-  all functionalities that this role shall be able to use in the admin interface or, for
-  functionalities that all registered users shall have access to, via the
-  ``functionalities.registered`` variable of your vars file.
+Add the layer names in a :ref:`integrator_finctionality` named ``filterable_layers````
```
Comment (jwkaltz) [inline doc/integrator/filter.rst]: typo: finctionality

File: doc/integrator/fulltext_search.rst [hunk 10]
```diff
@@ -10,34 +10,33 @@ The *text search* feature requires a dedicated PostgreSQL table. The full-text
 search table should be named ``tsearch`` (for *text search*) and should be in
 the application-specific schema.
 
-You do not need to create the table yourself, as it is created by the
-``create_db`` command line (see the section
-:ref:`integrator_install_application`).
+You do not need to create the table yourself, as it is created by the database creation
```
Comment (jwkaltz) [inline doc/integrator/fulltext_search.rst]: "as it is created by the database creation" -> "as it was already created during application installation"

File: doc/integrator/fulltext_search.rst [hunk 11]
```diff
@@ -265,15 +278,13 @@ instead of 'french'. For example:
...
 
     fulltextsearch:
         languages:
             fr: fr
 
-``fr: fr`` is a link between the pyramid language and
-the text search configuration, by default the it is
-``fr: french`` because the default french text search
-configuration is named 'french'.
+``fr: fr`` is a link between the pyramid language and the text search configuration, by default the it is
```
Comment (jwkaltz) [inline doc/integrator/fulltext_search.rst]: "the it is" -> "it is"

File: doc/integrator/functionality.rst [hunk 12]
```diff
@@ -1,3 +1,5 @@
+.. _integrator_fuctionality:
```
Comment (jwkaltz) [inline doc/integrator/functionality.rst]: typo "fuctionality" -> "functionality"

File: doc/integrator/functionality.rst [hunk 13]
```diff
@@ -87,68 +88,12 @@ The ``vars_<project>.yaml`` file includes variables for managing *functionalitie
...
-
-.. code:: javascript
+Usage in Javascript client
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    {
-        ptype: "cgxp_mapopacityslider",
-        defaultBaseLayerRef: "${functionality['default_basemap'][0] | n}"
-    }
+The functionnalities will be send to the Javascript client application throw the user web crebice.
```
Comment (jwkaltz) [inline doc/integrator/functionality.rst]: "functionnalities" -> "functionalities"

Comment (jwkaltz) [inline doc/integrator/functionality.rst]: "throw" -> "through"

Comment (jwkaltz) [inline doc/integrator/functionality.rst]: what is "crebice"?

File: doc/integrator/functionality.rst [hunk 14]
```diff
@@ -87,68 +88,12 @@ The ``vars_<project>.yaml`` file includes variables for managing *functionalitie
...
+To be published a functionalitis should be present in ``functionalities:available_in_templates`` parameter
```
Comment (jwkaltz) [inline doc/integrator/functionality.rst]: "functionalitis" -> "functionality"

File: doc/integrator/routing.rst [hunk 15]
```diff
@@ -3,29 +3,36 @@
 Routing interface
 =================
 
-This section describes how to add the `OSRM routing <http://project-osrm.org/>`_ interface to a c2cgeoprtal application.
+This section describes how to add the `OSRM routing <http://project-osrm.org/>`_ interface to a
+c2cgeoprtal application.
```
Comment (jwkaltz) [inline doc/integrator/routing.rst]: c2cgeoportal

File: doc/integrator/routing.rst [hunk 16]
```diff
@@ -81,14 +94,18 @@ profile: string, url-path of the profile, example::
         '                |     '     |    '
     backendUrl           |  profile  | osrm query
 
-If more than one profile are supplied, the component will show a drop-down-menu for the user to select a profile.
+If more than one profile are supplied, the component will show a drop-down-menu for the user to select a
```
Comment (jwkaltz) [inline doc/integrator/routing.rst]: "are supplied" -> "is supplied"

File: doc/integrator/tilegeneration.rst [hunk 17]
```diff
@@ -116,19 +112,16 @@ https://github.com/camptocamp/tilecloud-chain/blob/master/tilecloud_chain/scaffo
 The main thing to do is to:
 
 * Set the resolutions we want to generate in the ``grids``.
-  If we want to generate different resolution per layers we should create
-  deferent grid.
+  If we want to generate different resolution per layers we should create deferent grid.
```
Comment (jwkaltz) [inline doc/integrator/tilegeneration.rst]: "If we want to generate different resolutions par layer, we should create different grids."

File: doc/integrator/tilegeneration.rst [hunk 18]
```diff
@@ -116,19 +112,16 @@ https://github.com/camptocamp/tilecloud-chain/blob/master/tilecloud_chain/scaffo
...
   Sub-level of ``caches`` is the cache name.
 
-* Configure the ``layer_default``, the ``layers``, and the
-  ``generation``/``default_layers``.
+* Configure the ``layer_default``, the ``layers``, and the ``generation``/``default_layers``.
   Sub-level of ``layers`` is the layer name.
 
-* We can drop the empty tiles with an hash comparison,
-  tilecloud-chain has a tool to help us:
+* We can drop the empty tiles with an hash comparison, tilecloud-chain has a tool to help us:
```
Comment (jwkaltz) [inline doc/integrator/tilegeneration.rst]: "an hash" -> "a hash"

File: doc/integrator/tilegeneration.rst [hunk 19]
```diff
@@ -156,8 +149,7 @@ AWS generation.
 Tile Generation and management
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-This package offers two commands lines, one to generate the tiles locally,
-see help:
+This package offers two commands lines, one to generate the tiles locally, see help:
```
Comment (jwkaltz) [inline doc/integrator/tilegeneration.rst]: "offers two commands lines" -> "offers two command line tools"

File: doc/integrator/urllogin.rst [hunk 20]
```diff
@@ -16,8 +16,7 @@ The AES key must be either 16, 24, or 32 bytes long.
 
 To do that you should use the urllogin command: ``./docker-run urllogin --help``.
 
-It generate a token like: ``auth=148b60cc...`` that you can add it in the query string of
-the permalink.
+It generate a token like: ``auth=148b60cc...`` that you can add it in the query string of the permalink.
```
Comment (jwkaltz) [inline doc/integrator/urllogin.rst]: "It generate" -> "It generates"
"that you can add it" -> "that you can add"

File: doc/integrator/functionality.rst (hunk 13 above)
Comment (sbrunner) [reply doc/integrator/functionality.rst]: => service (just 3 faults in one word :-/ )

File: doc/integrator/cluster.rst (hunk 1 above)
Comment (jwkaltz) [reply doc/integrator/cluster.rst]: So suggestion:
"A customized version of the script, applicable for c2cgeoportal, is available in"

**Comment by sbrunner** [comment]
Many thanks @jwkaltz for your review, I just add a new small page here :-)
https://github.com/camptocamp/c2cgeoportal/pull/4487/files#diff-cdf8f8210db224019c722414fe6cf56a

File: doc/integrator/c2cwsgiutils.rst [hunk 21]
```diff
@@ -0,0 +1,10 @@
+.. _integrator_c2cwsgiutils
+
+c2cwsgiutils
+============
+
+c2cgeoportal use c2cwsgiutils you can need it for sone integrtion task, see his
```
Comment (jwkaltz) [inline doc/integrator/c2cwsgiutils.rst]: "c2cgeoportal" -> "c2cgeoportal uses"
"integrtion" -> "integration"
"his" -> "its"

But I am not sure what we are trying to say in the sentence? What kind of integration task can an integrator need it for?

File: doc/integrator/extend_data_model.rst [hunk 22]
```diff
@@ -1,55 +1,246 @@
...
+        ('interfaces', Interface),
+        ('userdetail', UserDetail),
+        ('title', Title),
+    ))
 
-        def __unicode__ (self):
-            return self.name or u''
+    with PermissionSetter(config):
+      # scan view decorator for adding routes
+      config.scan()
```
Comment (arnaud-morvan) [inline doc/integrator/extend_data_model.rst]: PermissionSetter not imported.
We want permission only on admin views.
```
with PermissionSetter(config):
    config.scan() = config.scan('<package>.admin.views')
config.scan(ignore='<package>.admin.views')
```

File: doc/integrator/ngeo.rst [hunk 23]
```diff
@@ -6,19 +6,17 @@ ngeo
 Organisation
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: Organi**z**ation

File: doc/integrator/caching.rst [hunk 24]
```diff
@@ -8,12 +8,10 @@ Headers
 
 By default most of the elements will be cached for 10 days by the browser.
 
-To change this value for all the application change in the
-``vars.yaml`` file the ``vars/default_max_age`` value. ``0`` will mean ``no-cache``.
-The specified value is in seconds.
+To change this value for all the application change in the ``vars.yaml`` file the ``vars/default_max_age``
```
Comment (arnaud-morvan) [inline doc/integrator/caching.rst]: ``vars/headers/<service_name>/cache_control_max_age`` value

File: doc/integrator/extend_data_model.rst [hunk 25]
```diff
@@ -1,55 +1,246 @@
...
+            }
+        })
+
+Now you need to extend the administration user interface.
+For this, first ensure that the following files exist (if needed, create them as empty files):
+
+* ``geoportal/<package>_geoportal/admin/__init__.py``:
+* ``geoportal/<package>_geoportal/admin/view/__init__.py``:
+
+the file ``geoportal/<package>_geoportal/admin/view/userdetail.py``:
```
Comment (arnaud-morvan) [inline doc/integrator/extend_data_model.rst]: admin/view => admin/views

File: doc/integrator/c2cwsgiutils.rst (hunk 21 above)
Comment (sbrunner) [reply doc/integrator/c2cwsgiutils.rst]: You can see all of them here: https://geomapfish-demo-dc.camptocamp.com/2.4/
short list:
- get packages versions
- changing log level
- memory diff
- enable SQL profiler
- ...

File: doc/integrator/extend_data_model.rst (hunk 22 above)
Comment (sbrunner) [reply doc/integrator/extend_data_model.rst]: Thanks Arnaud, but you probably want to say:
```
with PermissionSetter(config):
    config.scan('<package>.admin.views')
config.scan(ignore='<package>.admin.views')
```
right?

File: doc/integrator/c2cwsgiutils.rst (hunk 21 above)
Comment (jwkaltz) [reply doc/integrator/c2cwsgiutils.rst]: So suggestion:
"c2cgeoportal uses c2cwsgiutils. You can use functionalities from c2cwsgiutils to assist in integration tasks. See its documentation."

File: doc/integrator/caching.rst (hunk 24 above)
Comment (sbrunner) [reply doc/integrator/caching.rst]: Oups I should change this but booth are used but in defferent way...

Comment (sbrunner) [reply doc/integrator/caching.rst]: Just updated :-)

File: doc/integrator/extend_data_model.rst (hunk 22 above)
Comment (arnaud-morvan) [reply doc/integrator/extend_data_model.rst]: Yes right

File: doc/integrator/c2cwsgiutils.rst (hunk 21 above)
Comment (sbrunner) [reply doc/integrator/c2cwsgiutils.rst]: +1 thanks :-)

**Pull request merged**